src/components/MerchantsPage/gam/CampaignProgressBar.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import styled from 'styled-components';
+
+interface Props {
+  numContributions: number;
+  targetAmount: number;
+  progressBarColor: string;
+  lastContributionTime: Date;
+  endDate: Date;
+}
+
+const CampaignProgressBar = ({
+  numContributions,
+  targetAmount,
+  progressBarColor,
+  lastContributionTime,
+  endDate,
+}: Props) => {
+  const timeSinceLastContribution = Date.now() - lastContributionTime.getTime();
+  const minutesSinceLastConstribution = timeSinceLastContribution / (1000 * 60);
+  const hoursSinceLastContribution = minutesSinceLastConstribution / 60;
+  const daysSinceLastContribution = hoursSinceLastContribution / 24;
+
+  const progressWidth = (raised: number, target: number) => {
+    if (raised < target) return (raised / target) * 100;
+    return 100;
+  };
+  const progressPercent = Math.floor((numContributions / targetAmount) * 100);
+
+  const timeUntilEnd = endDate.getTime() - Date.now();
+  const daysUntilEnd = timeUntilEnd / (1000 * 3600 * 24);
+
+  return (
+    <ProgressBarContainer>
+      {hoursSinceLastContribution < 1 && (
+        <TimeStamp>
+          Last contribution made {Math.floor(minutesSinceLastConstribution)}m
+          ago
+        </TimeStamp>
+      )}
+      {hoursSinceLastContribution >= 1 && hoursSinceLastContribution < 24 && (
+        <TimeStamp>
+          Last contribution made {Math.floor(hoursSinceLastContribution)}h ago
+        </TimeStamp>
+      )}
+      {hoursSinceLastContribution >= 24 && (
+        <TimeStamp>
+          Last contribution made {Math.floor(daysSinceLastContribution)}d ago
+        </TimeStamp>
+      )}
+      <TargetAmountBar className="progress-bar">
+        <CurrentProgressBar
+          style={{
+            width: `${progressWidth(numContributions, targetAmount)}%`,
+            backgroundColor: progressBarColor,
+            //defaults to default color if no color is passed in
+          }}
+        >
+          {' '}
+        </CurrentProgressBar>
+      </TargetAmountBar>
+      <ContributionInfoContainer>
+        <div>
+          <ProgressTextContainer color={progressBarColor}>
+            {progressPercent}% to target
+          </ProgressTextContainer>{' '}
+          ({numContributions} out of {targetAmount} meals)
+        </div>
+        {daysUntilEnd > 1 && (
+          <div>
+            Ends in{' '}
+            <ProgressTextContainer color={progressBarColor}>
+              {Math.ceil(daysUntilEnd)}
```

Comment:
What happens if it goes over but the campaign is still active?

Reply:
<img width="785" alt="Screen Shot 2020-07-15 at 11 46 02 PM" src="https://user-images.githubusercontent.com/2313868/87624278-5eeb7d80-c6f5-11ea-9107-8ac935d6f28c.png">

Reply:
Hmm good question. Right now this will just say "Ended". Lets sync with Nanxi about this offline, this is a major edge case that warrants thought.

Reply:
My thinking is that if we want to keep a campaign active, we can just push out the end date in the DB? Then we don't hit this edge case

Reply:
Ok based on our Slack convo, gonna update this to display "0 days" if the campaign has reached its end date but is still active